File: bench/repos/mini_py_query_string_service_fixed/query.py
from urllib.parse import quote_plus
import unicodedata


def _canonical_scalar(value: object) -> str | None:
    if value is None or isinstance(value, (dict, list, tuple, set)):
        return None
    text = unicodedata.normalize("NFKD", str(value).strip()).encode("ascii", "ignore").decode("ascii")
    return text or None


def canonical_query(params: dict[str, object]) -> str:
    parts: list[str] = []
    for key in sorted(params):
        raw = params[key]
        values = raw if isinstance(raw, list) else [raw]
        for item in values:
            text = _canonical_scalar(item)
            if text is None:
                continue
            parts.append(f"{quote_plus(str(key))}={quote_plus(text)}")
    return "&".join(parts)

File: bench/repos/mini_ts_query_string_service_fixed/query.ts
function canonicalScalar(value: unknown): string | null {
  if (value == null || Array.isArray(value) || (typeof value === "object" && value !== null)) {
    return null;
  }
  const text = String(value).trim().normalize("NFKD").replace(/[\u0300-\u036f]/g, "");
  return text.length > 0 ? text : null;
}

export function canonicalQuery(params: Record<string, unknown>): string {
  const entries: string[] = [];
  for (const key of Object.keys(params).sort()) {
    const raw = params[key];
    const values = Array.isArray(raw) ? raw : [raw];
    for (const item of values) {
      const text = canonicalScalar(item);
      if (text == null) {
        continue;
      }
      entries.push(`${encodeURIComponent(key)}=${encodeURIComponent(text)}`);
    }
  }
  return entries.join("&");
}
